rules/windows/process_creation/win_app_windows_discovery.yml

```diff
@@ -0,0 +1,31 @@
+title: Application Window Discovery
+status: experimental
+description: Adversaries may attempt to get a listing of open application windows. Window listings could convey information about how the system is used or give context to information collected by a keylogger. In Mac, this can be done natively with a small AppleScript script - https://attack.mitre.org/techniques/T1155/
+author: Timur Zinniatullin, oscd.community
+references:
+    - https://attack.mitre.org/techniques/T1010/
+    - https://github.com/redcanaryco/atomic-red-team/blob/master/atomics/T1010/T1010.yaml
+logsource:
+    category: process_creation
+    product: windows
+detection:
+    selection1:
+        Image:
+            - '*\csc.exe'

```

Comment:
@zinint , as far as I understand related [Atomic Red Team test](https://github.com/redcanaryco/atomic-red-team/blob/master/atomics/T1010/T1010.yaml), it compiles some *code* that does the apps windows discovery. here you've constructed a rule that detects compilation of the code, not an actual execution and enumeration. am I right?

Reply:
As far as I get, it first compiles the code
```
**csc.exe -out:#{output_file_name} #{input_source_code}**
      #{output_file_name}
```
and then executes it to list main window titles (3rd arg)
```
csc.exe -out:#{output_file_name} #{input_source_code}
      **#{output_file_name}**
```

Perhaps I should wait on that and double check that with good people at atomicredteam.slack.com